lib/rspec/mocks/error_generator.rb

```diff
@@ -129,40 +129,35 @@ def raise_expired_test_double_error
       # @private
       def received_part_of_expectation_error(actual_received_count, *args)
         "received: #{count_message(actual_received_count)}" +
-          actual_method_call_args_description(actual_received_count, args)
+          method_call_args_description(args, actual_received_count, :actual)
       end
 
       # @private
       def expected_part_of_expectation_error(expected_received_count, expectation_count_type, argument_list_matcher)
         "expected: #{count_message(expected_received_count, expectation_count_type)}" +
-          expected_method_call_args_description(argument_list_matcher.expected_args)
+          method_call_args_description(argument_list_matcher.expected_args, expected_received_count, :expected)
       end
 
       # @private
-      def actual_method_call_args_description(count, args)
-        method_call_args_description(args) ||
-          if count > 0 && args.length > 0
-            " with arguments: #{args.inspect.gsub(/\A\[(.+)\]\z/, '(\1)')}"
-          else
-            ""
-          end
+      def show_method_call_args_description?(type, args, count)
+        if type == :actual
+          count > 0 && args.length > 0
+        elsif type == :expected
+          args.length > 0
+        end
       end
```

Comment:
Instead of defining this helper method and making callers of `method_call_args_description` pass the `count` and `type` args (neither of which are used in the description itself!), WDYT about making `method_call_args_description` yield to a block that would return a boolean value?  Then this logic could be moved into the appropriate caller.  I think that would be a bit cleaner.